### ISSUE TITLE
jnp.linalg.inv gave incorrect results

### ISSUE THREAD

**Issue by yyu22** [google/jax#14086]
### Description

I was trying to use ```jax.numpy.linalg.inv``` as a replacement for ```numpy.linalg.inv``` but found that ```jax.numpy.linalg.inv``` behaved differently from ```numpy.linalg.inv``` and gave incorrect results for matrix inversion. See below for a reproducible example:
```python
from jax import numpy as jnp
import numpy as np
from jax.numpy.linalg import inv as jnp_inv
from numpy.linalg import inv as np_inv

def kernel_np(X1, X2, l, sigma_f):
    X1 = np.expand_dims(X1, axis=1)
    X2 = np.expand_dims(X2, axis=1)
    sqdist = np.sum(X1 ** 2, 1).reshape(-1, 1) + np.sum(X2 ** 2, 1) - 2 * np.dot(X1, X2.T)
    return sigma_f ** 2 * np.exp(-0.5 / l ** 2 * sqdist)

def kernel_jnp(X1, X2, l, sigma_f):
    X1 = jnp.expand_dims(X1, axis=1)
    X2 = jnp.expand_dims(X2, axis=1)
    sqdist = jnp.sum(X1 ** 2, 1).reshape(-1, 1) + jnp.sum(X2 ** 2, 1) - 2 * jnp.dot(X1, X2.T)
    return sigma_f ** 2 * jnp.exp(-0.5 / l ** 2 * sqdist)

# np matrix inverse
X_train = np.linspace(0, 5, num=11)
Y_train = np.tanh(X_train)
K_np = kernel_np(X_train, X_train, 1.3773659885652723, 1.0147686098215838)
K_np_inv = np_inv(K_np)

# jnp matrix inverse
X_train_jnp = jnp.asarray(X_train)
Y_train_jnp = jnp.asarray(Y_train)
K_jnp = kernel_jnp(X_train_jnp, X_train_jnp, 1.3773659885652723, 1.0147686098215838)
K_jnp_inv = jnp_inv(K_jnp)
```
Multiplying the original matrix with its jnp inverse matrix does not return an identity matrix:
```
(K_jnp.dot(K_jnp_inv) - jnp.eye(K_jnp.shape[0])).max()
Out[19]: Array(1.5, dtype=float32)
```
but ```numpy.linalg.inv``` gave the correct results:
```
(K_np.dot(K_np_inv) - np.eye(K_np.shape[0])).max()
Out[21]: 1.6896939953329645e-09
```

### What jax/jaxlib version are you using?

jax-0.4.1, jaxlib-0.4.1

### Which accelerator(s) are you using?

GPU

### Additional system info

Python 3.9.5

### NVIDIA GPU info

```      
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 470.103.01   Driver Version: 470.103.01   CUDA Version: 11.4     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Tesla T4            Off  | 00000001:00:00.0 Off |                    0 |
| N/A   41C    P0    26W /  70W |   1350MiB / 15109MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
+-----------------------------------------------------------------------------+
```

**Comment by jakevdp** [comment]
Thanks for the question – this comes from the fact that JAX defaults to 32-bit computations which are more friendly on accelerators. If you [enable X64 mode](https://jax.readthedocs.io/en/latest/notebooks/Common_Gotchas_in_JAX.html#double-64bit-precision) before running your computation, you'll see that the JAX accuracy is consistent with that of NumPy:
```python
>>> (K_jnp.dot(K_jnp_inv) - jnp.eye(K_jnp.shape[0])).max()
DeviceArray(1.14232535e-09, dtype=float64)

>>> (K_np.dot(K_np_inv) - np.eye(K_np.shape[0])).max()
1.6896939953329645e-09
```

**Comment by yyu22** [comment]
Thanks! That worked.
